Framework/Sources/FloatingPanel.swift

```diff
@@ -477,24 +520,98 @@ class FloatingPanel: NSObject, UIGestureRecognizerDelegate {
-        preserveContentVCLayoutIfNeeded()
+        // Moving the finger upwards on the screen results in a negative Y translation. But since the animations are
+        // defined to progress when moving from bottom to top, we have to negate the gesture recognizer translation
+        // to get a positive animation completion fraction.
+        guard let panGestureContext = self.panGestureContext else {
+            fatalError("The pan gesture context needs to be set at the beginning of the pan")
+        }
+        var fraction = (-1 * (translation.y - (panGestureContext.alreadyMovedTranslationOfPan.y)) / panGestureContext.totalYTranslation)
 
-        let didMove = (preY != currentY)
-        guard didMove else { return }
+        // In order to still use the correct value for the fraction (-> value between 0 and 1) when the panel is moved
+        // downwards we need to multiply it by -1.
+        if (self.state == .full && panGestureContext.nextState == .half) || (self.state == .half && panGestureContext.nextState == .tip) {
+            fraction *= -1
+        }
 
-        if let vc = viewcontroller {
-            vc.delegate?.floatingPanelDidMove(vc)
+        let previousFractionComplete = animator.fractionComplete
+
+        // Manually set the fractionComplete value of the animator so the progress of animator corresponds with the
+        // drag on the floating panel as well as its current position on the screen.
+        animator.fractionComplete = fraction
+
+        // When the floating panel transitions from one state to another we need to also start a new animation.
+        // Therefor, we need to check whether the currenlty running animator is either finished OR got back to its
+        // starting point.
+        // There are two cases here since the user could drag the panel e.g. from .tip over .half to .full but change
+        // directions midway between .half and .full and drags the panel back downards.
+        if (animator.fractionComplete == 1) || (animator.fractionComplete == 0 && self.panGestureContext?.nextState != self.state) {
+            self.state = (animator.fractionComplete == 0) ? self.state : self.panGestureContext?.nextState ?? self.state
+            let newTargetState = self.nextPosition(with: velocity)
+            let distance = self.distance(to: newTargetState)
+
+            if newTargetState != self.panGestureContext?.nextState && newTargetState != self.state {
+                self.startNewAnimator(for: distance, with: velocity, movingTo: newTargetState)
+            }
+
+            // Since the drag is already in progress we need to 'save' the already travelled distance on the screen in
+            // order to calculate the correct translation when calcualting the fraction (-> progress) of the animator.
+            // Otherwise the panel 'jumps' from its current position to the finger's position on the screen.
+            self.panGestureContext?.alreadyMovedTranslationOfPan = translation
```

Comment:
Is it really correct to only update this in case the gesture is at completion fraction `0` or `1`?

Reply:
The assumption was that only if a new animation is started the `alreadyMovedTranslationOfPan` is of interest since we want to avoid seeing the panel "jump".